src/libcrun/seccomp_notify.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * crun - OCI runtime written in C
+ *
+ * Copyright (C) 2020 Giuseppe Scrivano <giuseppe@scrivano.org>
+ * crun is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
```

Comment:
Can we have the list of allowed and disallowed licenses for plugins?

Reply:
I am not a lawyer, but LGPL allows being used with proprietary software as long as changes to  the program itself are distributed.  In this case only changes to libcrun matter

Reply:
IANAL either, but as crun CLI is licensed under the terms of GPL2-or-later, I'm wondering that plugins might need to be GPL-compatible

Reply:
An alternative design is to implement the plugin infra on a standalone receiver binary with a more permissive license, rather than on crun itself.

Reply:
the idea is to use the same plugins from conmon, since it is already a long running process.

Having them supported also in crun makes just easier to debug stuff.

Reply:
> **When is a program and its plug-ins considered a single combined program?**
> ...
> If the main program dynamically links plug-ins, and they make function calls to each other and share data structures, we believe they form a single combined program, which must be treated as an extension of both the main program and the plug-ins. 

https://www.gnu.org/licenses/gpl-faq.en.html#GPLPlugins


> **If I write a plug-in to use with a GPL-covered program, what requirements does that impose on the licenses I can use for distributing my plug-in?**
> ...
> If the main program and the plugins are a single combined program then this means you must license the plug-in under the GPL or a GPL-compatible free software license and distribute it with source code in a GPL-compliant way. A main program that is separate from its plug-ins makes no requirements for the plug-ins.

https://www.gnu.org/licenses/gpl-faq.en.html#GPLAndPlugins

 - - -

If the plugin infra is directly implemented in the main crun binary, it is highly likely that the crun and the plugins are "a single combined program" and hence the plugins need to be GPL-compatible.

If the plugins need to be GPL-compatible, it is likely that the plugins cannot be used by runc, which is licensed under the terms of Apache License 2.0 ([incompatible with GPL2](https://www.apache.org/licenses/GPL-compatibility.html)), but IANAL.

cc @cyphar

Reply:
MIT/BSD/et al or MPL-2.0 licensed plugins would be compatible with both Apache-2.0 and LGPL-2.1, and thus wouldn't cause any issues.

But I would further note that you linked the GPL FAQ -- the LGPL treats dynamically shared libraries differently to the GPL, and dynamic plugins (especially of the form used here) are functionally identical to shared libraries. For a more famous example, the Name Service Switch (which is supported by glibc -- an LGPL project) supports using dynamic plugins. Those plugins do not need to be LGPL-3.0 licensed. So I would argue that it is not the case that plugins would need to be LGPL licensed.